### PR TITLE
Add caption prop to the Table component

### PR DESCRIPTION
**Pull request by ianoxley**
The `Table` component now supports an optional `caption` prop which, if given, renders as a `<caption>` element within the table.

## Related Issues / Pull Requests
#1143 

## Screenshot

VoiceOver rotor showing tables with captions:

rotor_tables

## Testing Instructions
The `caption` is visually hidden but can be tested using:

* Browser dev tools e.g. element inspector
* VoiceOver on OS X - turn on with Cmd+F5, bring up the rotor using Ctrl+Option+U, use left / right arrows to find the Tables, and you should see the table listed via its caption.

### CODE REVIEW

**Comment by ianoxley** [comment]
Updated the CSS used to hide the caption to use the `visually-hidden` mixin.

File: src/components/table/table.js [hunk 1]
```diff
@@ -1005,6 +1013,21 @@ class Table extends React.Component {
...
 
   /**
+   * Returns the caption prop wrapped in a <caption> tag,
+   * or null if no caption prop was given.
+   *
+   * @method caption
+   * @return {Object} JSX
+   */
+  get caption() {
+    if (this.props.caption) {
```
Comment (abramin) [inline src/components/table/table.js]: Our best practice advice would be to not use getters. The file is already full of them I know, so not a blocker.

File: src/components/table/table.scss [hunk 2]
```diff
@@ -26,6 +27,14 @@
   }
 }
 
+.carbon-table__caption {
+  // Discussed with KM and for now captions can be hidden from view so they
+  // don't impact the design, but still available to screen readers, etc.
```
Comment (abramin) [inline src/components/table/table.scss]: Do we want to immortalise KM like this?

File: src/components/table/__spec__.js [hunk 3]
```diff
@@ -931,6 +930,21 @@ describe('Table', () => {
       expect(parent.className).toEqual('carbon-table foo');
     });
 
+    it('renders a caption tag when a caption prop is given', () => {
+      let wrapper = render(
```
Comment (toni-leigh) [inline src/components/table/__spec__.js]: why is `render` required here?

Comment (ianoxley) [inline src/components/table/__spec__.js]: Because I'm only interested in verifying that a `caption` is included in the rendered HTML.

Comment (toni-leigh) [inline src/components/table/__spec__.js]: So you need to use `render` for this case? Would this work with `shallow` - curious

Comment (ianoxley) [inline src/components/table/__spec__.js]: I can't remember to be honest. I suspect it didn't work with `shallow` when I tried it, but it works fine with `shallow` now.

Also, I've just compared using `render` to using `shallow`, and `shallow` was 4-5 times faster! So I've gone with `shallow` 👍

Comment (toni-leigh) [inline src/components/table/__spec__.js]: Interesting, I'#ve never actually bench-marked them, but I think `shallow` is the fastest from what I've read, that's it's main benefit

**Pull request merged**